pkg/whiteblacklist/whiteblacklist.go

```diff
@@ -85,13 +86,20 @@ func (l *WhiteBlackList) Exclude(items []string) {
 
 // IsIncluded returns if the given item is included.
 func (l *WhiteBlackList) IsIncluded(item string) bool {
-	_, exists := l.list[item]
+	var matched bool
+	for whiteListPattern := range l.list {
+		var err error
+		matched, err = regexp.MatchString(whiteListPattern, item)
+		if matched && err == nil {
```

Comment:
I don't feel too great about the `err` check here. An alternative here would be validating the regex pattern (whether it's parseable or not) beforehand. That way, we wouldn't need to account for `err` check and use something like `regexp.MustCompile`

Reply:
yeah I think we should validate these at startup and fail immediately if not a valid regex

Reply:
This has been done.

Reply:
Doing this in two places seems error prone. Could we architect it in a way where we do the parsing just once and create the whitelist/blacklist filter at that point already?

Reply:
I hear you. I did think of rearchitecting this. Might as well do it in this PR.